api/pb/types.proto

```diff
@@ -14,10 +14,14 @@ message Pin {
   repeated bytes Allocations = 3;
   sint32 MaxDepth = 4;
   bytes Reference = 5;
-  reserved 6,7;
-  sint32 ReplicationFactorMin = 8;
-  sint32 ReplicationFactorMax = 9;
-  string Name = 10;
-  uint64 ShardSize = 11;
-  map<string, string> Metadata = 12;
+  PinOptions Options = 6;
+}
+
+message PinOptions {
+  sint32 ReplicationFactorMin = 1;
+  sint32 ReplicationFactorMax = 2;
+  string Name = 3;
+  uint64 ShardSize = 4;
+  repeated string UserAllocations = 5;
```

Comment:
@hsanjuan What is the current strategy for when the allocations that a user has specified are no longer available? I am not sure that this is worth keeping around long term?

Reply:
Nothing happens. Things will get preferentially assigned to those allocations and we keep them as an indicator of where things were meant to be pinned in the first place. Will this data ever be re-used for something, I'm not sure at this point...

I think we should keep the field in the protobuf, but we might want to simply not use it (never setting it). But in most cases it won't be used anyway.

Reply:
Thinking again, if we're not going to use it, we may remove it from the protobuf and add it when we use it.